pkg/gofr/datasource/file/ftp/interface.go

```diff
@@ -102,12 +97,8 @@ type FileSystemProvider interface {
 }
 
 // Logger interface is used by ftp package to log information about query execution.
-// Developer Notes: Note that it's a reduced version of logging.Logger interface. We are not using that package to
-// ensure that ftp package is not dependent on logging package. That way logging package should be easily able
-// to import ftp package and provide a different "pretty" version for different log types defined here while
-// avoiding the cyclical import issue. Idiomatically, interfaces should be defined by packages who are using it; unlike
-// other languages. Also - accept interfaces, return concrete types.
 type Logger interface {
-	Logf(format string, args ...interface{})
-	Errorf(format string, args ...interface{})
+	Debugf(string, string, string, ...interface{})
+	Logf(string, string, string, ...interface{})
+	Errorf(string, string, string, ...interface{})
```

Comment:
If you’re going to have a number of sequential `string` arguments, it can help to name them: `operation, status, msg string` for example.